File: frontend/src/types/siteSource.ts
// 站点来源表 site_source 字段定义
// 表设计: 网站名称 / 网址 / 图片规则 / 漫画详情网址 / 漫画内容网址

export interface SiteSource {
  /** 主键ID */
  id: number
  /** 网站名称 */
  name: string
  /** 网址(站点根域名) */
  url: string
  /** 图片规则(图片URL解析/替换规则, 支持占位符说明) */
  imageRule: string
  /** 漫画详情网址模板(可含 {id} / {bookId} 占位符) */
  detailUrl: string
  /** 漫画内容网址模板(章节阅读页, 可含 {id} / {chapterId} 占位符) */
  contentUrl: string
  /** 启用状态(后端原始: 1启用 0停用) */
  enabled?: number
  /** 启用状态(派生布尔, 便于 UI 使用) */
  isEnabled?: boolean
  /** 备注 */
  remark?: string
  /** 创建时间 */
  createdAt?: string
  /** 更新时间 */
  updatedAt?: string
}

// 新增/编辑表单载荷(不含 id 与时间戳), enabled 以数字提交(1/0)
export type SiteSourcePayload = Omit<SiteSource, 'id' | 'createdAt' | 'updatedAt' | 'isEnabled'>

// 查重结果(全站查重: 当前漫画表中未匹配的漫画名称)
export interface DedupResult {
  bookId: number
  bookName: string
  sourceSite: string
  matchedSiteName?: string
}

File: backend/src/main/java/com/man3/entity/SiteSource.java
package com.man3.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 漫画来源站点表
 */
@Data
@TableName("site_source")
public class SiteSource {

    /** 主键ID */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 网站名称 */
    private String name;

    /** 网址(站点根域名) */
    private String url;

    /** 图片规则(图片URL解析/替换规则说明) */
    private String imageRule;

    /** 漫画详情网址模板(支持{bookId}/{id}占位符) */
    private String detailUrl;

    /** 漫画内容(阅读页)网址模板(支持{chapterId}/{id}占位符) */
    private String contentUrl;

    /** 是否启用(1启用 0停用) */
    private Integer enabled;

    /** 备注 */
    private String remark;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}
